src/crepe/api/Button.h

```diff
@@ -0,0 +1,88 @@
+#pragma once
+
+#include <functional>
+
+#include "UiObject.h"
+
+namespace crepe {
+
+/**
+ * \class Button
+ * \brief Represents a clickable UI button, derived from the UiObject class.
+ * 
+ * This class provides functionality for a button in the UI, including toggle state,
+ * click handling, and mouse hover detection. A callback function can be provided to
+ * handle button clicks.
+ */
+class Button : public UiObject {
+public:
+	/**
+     * \brief Constructs a Button with the specified game object ID and dimensions.
+     * 
+     * \param id The unique ID of the game object associated with this button.
+     * \param width The width of the button.
+     * \param height The height of the button.
+     * \param is_toggle Optional flag to indicate if the button is a toggle button. Defaults to false.
+     * \param on_click callback function that will be invoked when the button is clicked.
+     */
+	Button(game_object_id_t id, int width, int height, std::function<void()> on_click,
```

Comment:
`width` en `height` zijn integers, maar in de rest van de engine worden width/height en x/y samengebundeld dmv `vec2` (of `ivec2` in dit geval). Zijn deze getallen in world units, en zo ja, waarom zijn dit integers en geen floats?